tests/unit/test_athena_iceberg.py

```diff
@@ -924,3 +924,56 @@ def test_to_iceberg_uppercase_columns(
     )
 
     assert_pandas_equals(df, df_output)
+
+
+def test_to_iceberg_fill_missing_columns_map(
+    path: str,
+    path2: str,
+    glue_database: str,
+    glue_table: str,
+) -> None:
+    df_with_col = pd.DataFrame(
+        {
+            "partition": [1, 1, 2, 2],
+            "column2": ["A", "B", "C", "D"],
+            "map_col": [{"s": "d"}, {"s": "h"}, {"i": "l"}, {}],
```

Comment:
Hi, could we test this for struct fields as well? I have similar issues with struct<a: string, struct<b: string>>, failing with the exact same error.

Reply:
A example column could be 
`
'struct_col': [{'attr1': 'val1', 'attr2': {'attr21': 'val21'}},
                       {'attr1': 'val1', 'attr2': {'attr21': None}},
                       {'attr1': 'val1', 'attr2': None},
                       {},
                       ]
`